gusto/time_discretisation/explicit_runge_kutta.py

```diff
@@ -379,11 +379,8 @@ def solve_stage(self, x0, stage):
 
             # Update field_i for physics / limiters
             for evaluate in self.evaluate_source:
-                # TODO: not implemented! Here we need to evaluate the m-th term
                 # in the i-th RHS with field_m
-                raise NotImplementedError(
-                    'Physics not implemented with RK schemes that use the '
-                    + 'predictor form')
+                evaluate(self.field_i[stage], self.dt)
```

Comment:
I think this is correct, unless why understanding of evaluate is wrong.. We evaluate the physics terms in `field_i` from the previous stage, since `field_i` is a list we have all previous states for each stage. The physics terms (non-time derivative) are then in the `rhs` that goes into the solver.

Reply:
So the equation we are trying to solve is:
```
y^m = y^0 - dt*Sum_{i=0}^{m-1} a_{m,i} * F[y^i], for m = 1 to M - 1
```

What the `evaluate` method does is perform `F[y]` and put the value of that into a source field `S`. The thing that we don't have is `M` different source fields -- we only have one. So rather than having `M` source fields containing `F[y^i]` (for i=1 to m-1), and doing `m-1` evaluations, we are overwriting `S` with ``F[y^{m-1}]` and not bringing in all of the correct contributions.

I don't think coding this is straightforward which is why I think it hasn't been done yet! But hopefully a test could demonstrate what is correct or not...